lab-lee/test/duck-route-test.js

```diff
@@ -0,0 +1,260 @@
+          .set('Content-Type', 'application/json')
+          .end((err, res) => {
+            expect(res.status).to.equal(400);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  describe('testing GET requests', function() {
+
+    before( done => {
+      exampleFowl.timestamp = new Date();
+      new Fowl(exampleFowl).save()
+      .then( fowl => {
+        this.tempFowl = fowl;
+        return Fowl.findByIdAndAddDuck(fowl._id, exampleDuck);
+      })
+      .then( duck => {
+        this.tempDuck = duck;
+        done();
+      })
+      .catch(done);
+    });
+
+    after( done => {
+      Promise.all([
+        Fowl.remove({}),
+        Duck.remove({}),
+      ])
+      .then(() => done())
+      .catch(done);
+    });
+
+    describe('with valid duck id', () => {
+
+
+      it('should return a duck', done => {
+        request.get(`${url}/api/duck/${this.tempDuck._id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal(exampleDuck.name);
+          expect(res.body.fowlID).to.equal(this.tempFowl._id.toString());
+          done();
+        });
+      });
+
+      describe('testing GET requests with invalid duck id', () => {
+
+        it('should return a 404 not found', done => {
+          request.get(`${url}/api/duck/noduck`)
+          .end((err, res) => {
+            expect(res.status).to.equal(404);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  describe('testing DELETE requests', function() {
+
+    before( done => {
+      exampleFowl.timestamp = new Date();
+      new Fowl(exampleFowl).save()
+      .then( fowl => {
+        this.tempFowl = fowl;
+        return Fowl.findByIdAndAddDuck(fowl._id, exampleDuck);
+      })
+      .then( duck => {
+        this.tempDuck = duck;
+        done();
+      })
+      .catch(done);
+    });
+
+    after( done => {
+      delete exampleFowl.timestamp;
+      if(this.tempFowl) {
+        Fowl.remove({})
+        .then(() => done())
+        .catch(done);
+        return;
+      }
+      done();
+    });
+
+    describe('with valid duck id', () => {
+
+      it('should delete a duck', done => {
+        request.delete(`${url}/api/duck/${this.tempDuck._id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(204);
+          done();
+        });
+      });
+
+      describe('testing DELETE requests with invalid duck id', () => {
+
+        it('should return a 404 not found', done => {
+          request.delete(`${url}/api/duck/noduck`)
+          .end((err, res) => {
+            expect(res.status).to.equal(404);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  describe('Testing PUT /api/duck:id requests', function() {
+
+    before( done => {
+      exampleFowl.timestamp = new Date();
+      new Duck(exampleDuck).save()
```

Comment:
I'm surprised this worked and Mongo didn't complain that you created a duck without first creating a fowl.